src/lib/profiles/data-management/Current/LoggingManagement.cpp

```diff
@@ -475,59 +447,57 @@ WEAVE_ERROR LoggingManagement::SetExchangeManager(nl::Weave::WeaveExchangeManage
  * @brief
  *   LoggingManagement constructor
  *
- * For prioritization to work correctly, inBuffers must be incrementally
- * increasing in priority.
+ * Initialize the LoggingManagement with an array of LogStorageResources.  The
+ * array must provide a resource for each valid importance level; the first
+ * element in the array corresponds to ImportanceType::ProductionCritical and
+ * the last element of the array corresponds to ImportanceType::kImportanceType_Last.
  *
  * @param[in] inMgr         WeaveExchangeManager to be used with this logging subsystem
  *
- * @param[in] inNumBuffers  Number of buffers to use for event storage
- *
- * @param[in] inBufferLengths Description of inBufferLengths
+ * @param[in] inLogStorageResources  An array of LogStorageResources for each importance level.
  *
- * @param[in] inBuffers     The buffers to use for actual event logging.
- *
- * @param[in] inCounterKeys Keys naming persisted counters
- *
- * @param[in] inCounterEpochs An array of epochs for each of the persisted counters
- *
- * @param[in] inCounterStorage Application-provided storage for the persistent counters
- *
- * @return LoggingManagement
  */
-LoggingManagement::LoggingManagement(WeaveExchangeManager * inMgr, size_t inNumBuffers, size_t * inBufferLengths, void ** inBuffers,
-                                     nl::Weave::Platform::PersistedStorage::Key * inCounterKeys, const uint32_t * inCounterEpochs,
-                                     nl::Weave::PersistedCounter ** inCounterStorage)
+
+LoggingManagement::LoggingManagement(nl::Weave::WeaveExchangeManager * inMgr,
+                                     const LogStorageResources * const inLogStorageResources)
 {
     CircularEventBuffer * current = NULL;
     CircularEventBuffer * prev    = NULL;
     CircularEventBuffer * next    = NULL;
-    size_t i;
+    size_t i, j;
+
+    size_t numResources = (kImportanceType_Last - kImportanceType_First) + 1;
 
     mThrottled   = 0;
     mExchangeMgr = inMgr;
 
-    for (i = 0; i < inNumBuffers; i++)
+    for (j = 0; j < numResources; j++)
     {
-        next = ((i + 1) < inNumBuffers) ? static_cast<CircularEventBuffer *>(inBuffers[i + 1]) : NULL;
+        i = (kImportanceType_Last - kImportanceType_First) - j;
+
+        next = (i > 0) ? static_cast<CircularEventBuffer *>(inLogStorageResources[i - 1].mBuffer) : NULL;
 
-        new (inBuffers[i]) CircularEventBuffer(static_cast<uint8_t *>(inBuffers[i]) + sizeof(CircularEventBuffer),
-                                               inBufferLengths[i] - sizeof(CircularEventBuffer), prev, next);
+        new (inLogStorageResources[i].mBuffer)
+            CircularEventBuffer(static_cast<uint8_t *>(inLogStorageResources[i].mBuffer) + sizeof(CircularEventBuffer),
+                                inLogStorageResources[i].mBufferSize - sizeof(CircularEventBuffer), prev, next);
```

Comment:
Should we check appropriate buffer size:
    VerifyOrExit(inLogStorageResources[i].mBufferSize > sizeof(CircularEventBuffer), err = WEAVE_ERROR_BUFFER_TOO_SMALL)

Reply:
Excellent idea, Done.  One small change I went with is use `VerifyOrDie` as the function returns no error, and there is no reasonable recovery from such an error